file/file.go

```diff
@@ -0,0 +1,70 @@
+// Package file - Handle file operations
+package file
+
+import (
+	"encoding/json"
+	"errors"
+	"os"
+)
+
+const filename = "tasks.json"
+
+// File - file generic struct. T generic.
+type File[T any] struct{}
+
+// New - instantiates a File struct.
+func New[T any]() *File[T] {
+	return &File[T]{}
+}
+
+// Write - writes tasks into a JSON file
+func (f *File[T]) Write(data *[]T) error {
```

Comment:
As we removed T as unnecessary, File struct no longer needed at all, it can be just functions. So the package is responsible for storage format and logic. The logic is stateless, package is small, so no need to introduce one more entity that doesn't help.